src/telemetry/telemetry.c

```diff
@@ -445,6 +449,46 @@ add_function_call_telemetry(JsonbParseState *state)
 	pushJsonbValue(&state, WJB_END_OBJECT, NULL);
 }
 
+static void
+add_replication_telemetry(JsonbParseState *state)
+{
+	int res;
+	bool isnull;
+	Datum data;
+
+	if (SPI_connect() != SPI_OK_CONNECT)
+		return;
+
+	res = SPI_execute("SELECT count(pid) from pg_catalog.pg_stat_get_wal_senders() WHERE pid is "
+					  "not null",
```

Comment:
`COUNT(<column>)` usually [counts only non-null columns](https://www.postgresql.org/docs/14/functions-aggregate.html) so don't think that you need the `not null`.

```suggestion
	res = SPI_execute("SELECT count(pid) from pg_catalog.pg_stat_get_wal_senders()",
```

Reply:
Is this query actually giving us the thing we are looking for? Can't there be many different kinds of WAL senders, e.g., for backups, archiving, logical replication, etc? Do we want all of those   or do we want to try and categorizing them or at least determine which ones are actual standby servers?

Reply:
@mkindahl hmm, that was orginally a holdover from a `bool_and()` but looking at it now I think I like that it's explicit "we're counting all the rows where pid is not null", and not that we're counting pids and are missing null ones by oversight. If I was to do a change for this I might prefer to do
```sql
SELECT count(*) FROM pg_catalog.pg_stat_get_wal_senders() WHERE pid is not null
```

Reply:
@erimatnor unfortunately there doesn't seem to really be a great way to tell them apart at this layer¹, though fortunately we should generally have the replicas in telemetry also to correlate with. I can rename the JSON field to `num_wal_senders` if you perfer to make what this is sending clearer. 

¹For instance, `pg_stat_replication` is also just returning all the rows from this function.